engine/balances.go

```diff
@@ -696,31 +696,35 @@ func (b *Balance) AsBalanceSummary(typ string) *BalanceSummary {
 }
 
 func (b *Balance) Publish() {
-	if b.account == nil ||
-		thresholdS == nil {
+	if b.account == nil {
 		return
 	}
 	accountId := b.account.ID
 	acntTnt := utils.NewTenantID(accountId)
-	thEv := &ArgsProcessEvent{
-		CGREvent: utils.CGREvent{
-			Tenant: acntTnt.Tenant,
-			ID:     utils.GenUUID(),
-			Event: map[string]interface{}{
-				utils.EventType:   utils.BalanceUpdate,
-				utils.EventSource: utils.AccountService,
-				utils.Account:     acntTnt.ID,
-				utils.BalanceID:   b.ID,
-				utils.Units:       b.Value}}}
+	cgrEv := utils.CGREvent{
+		Tenant: acntTnt.Tenant,
+		ID:     utils.GenUUID(),
+		Event: map[string]interface{}{
+			utils.EventType:   utils.BalanceUpdate,
+			utils.EventSource: utils.AccountService,
+			utils.Account:     acntTnt.ID,
+			utils.BalanceID:   b.ID,
+			utils.Units:       b.Value}}
 	if !b.ExpirationDate.IsZero() {
-		thEv.Event[utils.ExpiryTime] = b.ExpirationDate.Format(time.RFC3339)
+		cgrEv.Event[utils.ExpiryTime] = b.ExpirationDate.Format(time.RFC3339)
+	}
+	if statS != nil {
+		var reply []string
+		go statS.Call(utils.StatSv1ProcessEvent, &StatsArgsProcessEvent{CGREvent: cgrEv}, &reply)
```

Comment:
here should be better to use anonymous function and define reply inside. Also I see you are not treating the error (log it to syslog)